AnimView/th.cpp

```diff
@@ -374,7 +378,8 @@ void THAnimations::drawFrame(wxImage& imgCanvas, size_t iAnimation,
   int iFarX = 0;
   int iFarY = 0;
   while ((pElement = _getElement(iListIndex++))) {
-    if (pMask != NULL && !pMask->isSet(pElement->flags >> 4, pElement->layerid))
+    if (pMask != nullptr &&
+        !pMask->isSet(pElement->flags >> 4, pElement->layerid))
```

Comment:
field access could be moved into `pElement`

Reply:
Right now th_element_t and a number of other structs in this class have exact sizing and layout requirements (#pragma pack(1)) making them difficult to extend, and to some extent implementation defined behaviour.

We do the same thing in CorsixTH proper but we immediately copy the properties on to another non-packed struct after loading. (CorsixTH/Src/th_gfx.cpp:253)

Room for improvements there.